tests/RandomHelpersT.cpp

```diff
@@ -41,10 +47,98 @@ TEST_CASE("[Random] Fast random generation")
 
     REQUIRE(test(0.0f, 1.0f, numGenerations, numDivisions));
     REQUIRE(test(-1.0f, 1.0f, numGenerations, numDivisions));
     REQUIRE(test(0.0f, 123.0f, numGenerations, numDivisions));
     REQUIRE(test(-123.0f, 0.0f, numGenerations, numDivisions));
 }
+
+/**
+ * @brief Check the behavior of a gaussian real random generator
+ *
+ * - compute a histogram of random generations
+ * - compare it against the standard library normal distribution
+ */
+template <unsigned Quality>
+static bool gaussianRandomTest(double mean, float variance, size_t numGen, size_t histSize, double maxAbsErr)
+{
+    fast_gaussian_generator<float, Quality> gen(mean, variance);
+
+    std::mt19937_64 prng;
+    std::normal_distribution<float> dist(mean, variance);
+
+    // the tested bounds
+    const double min = -1.0 + mean;
+    const double max = +1.0 + mean;
+
+    // generate, quantify, count occurrences
+    std::vector<size_t> counts(histSize);
+    for (size_t i = 0; i < numGen; ++i) {
+        double value = gen();
+        double normalized = (value - min) / (max - min);
+        long bin = std::lround(histSize * normalized);
+        if (bin >= 0 && static_cast<size_t>(bin) < histSize)
+            counts[bin] += 1;
+    }
+    std::vector<size_t> referenceCounts(histSize);
+    for (size_t i = 0; i < numGen; ++i) {
+        double value = dist(prng);
+        double normalized = (value - min) / (max - min);
+        long bin = std::lround(histSize * normalized);
+        if (bin >= 0 && static_cast<size_t>(bin) < histSize)
+            referenceCounts[bin] += 1;
+    }
+
+    // bin probabilities
+    std::vector<double> proba(histSize);
+    std::vector<double> referenceProba(histSize);
+    for (size_t i = 0; i < histSize; ++i) {
+        proba[i] = static_cast<double>(counts[i]) / numGen;
+        referenceProba[i] = static_cast<double>(referenceCounts[i]) / numGen;
+    }
+
+    // normalize to unity, for the sake of comparison
+    {
+        double sum = 0.0;
+        for (double x : referenceProba)
+            sum += x;
+        double scaleFactor = 1.0 / sum;
+        for (size_t i = 0; i < histSize; ++i) {
+            proba[i] *= scaleFactor;
+            referenceProba[i] *= scaleFactor;
+        }
+    }
+
+    auto normalizeToUnity = [](absl::Span<double> v) {
+        double s = 0;
+        for (double x : v)
+            s += x;
+        for (double& x : v)
+            x *= 1.0 / s;
+    };
+    normalizeToUnity(absl::MakeSpan(proba));
+    normalizeToUnity(absl::MakeSpan(referenceProba));
+
+    // compare
+    for (size_t i = 0; i < histSize; ++i) {
+        double absErr = std::abs(proba[i] - referenceProba[i]);
+        if (absErr > maxAbsErr)
+            return false;
+    }
+
+    return true;
+}
+
+TEST_CASE("[Random] Gaussian random generation")
+{
+    unsigned numGenerations = 16384;
+    unsigned numDivisions = 128;
+
+    const double maxAbsErr = 0.05; // PDF ±5%
```

Comment:
This is one of my issues with probability tests. This could always fail in some cases, especially since we have both the observed and expected frequencies as random. I wonder if we should make this test more "human based", e.g. plotting histograms and deeming the results "good enough"?

Reply:
The generator is performed always with the same seeds, so this does not run at risk that successive runs of the same tests will fail. If it happens by some chance, increase the number of random picks.
It's already chosen to be some magnitudes above the number of bins.

Reply:
OK!